### PR TITLE
✨ [RUM 5983] Add set view context apis

### PR DESCRIPTION
**Pull request by cy-moi**
## Motivation
Add `setViewContext()` and `setViewContextProperty()` public APIs. 

## Changes
- Rename `ViewContexts` to `ViewHistory` for future clarity
- Add `setViewContext()` and `setViewContextProperty()` public APIs
- Add tests for new APIs

## Testing

- [x] Local
- [x] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/contexts/viewHistory.spec.ts [hunk 1]
```diff
@@ -154,10 +163,28 @@ describe('viewContexts', () => {
...
-      expect(viewContexts.findView(targetTime)).toBeDefined()
+      expect(viewHistory.findView(targetTime)).toBeDefined()
 
       clock.tick(VIEW_CONTEXT_TIME_OUT_DELAY + CLEAR_OLD_VALUES_INTERVAL)
-      expect(viewContexts.findView(targetTime)).toBeUndefined()
+      expect(viewHistory.findView(targetTime)).toBeUndefined()
+    })
+  })
+
+  describe('custom context', () => {
```
[the remark this answers is not in the document]
Comment (cy-moi) [reply packages/rum-core/src/domain/contexts/viewHistory.spec.ts]: 🙏 Removed.

**Review by nulrich — APPROVED** [review]

**Review by thomas-lebeau — APPROVED** [review]

File: packages/rum-core/src/domain/view/trackViews.ts [hunk 2]
```diff
@@ -202,6 +211,8 @@ function newView(
   let documentVersion = 0
   let endClocks: ClocksState | undefined
   const location = shallowClone(initialLocation)
+  const viewCustomerDataManager = createCustomerDataTrackerManager()
+  const contextManager = createContextManager(viewCustomerDataManager.getOrCreateTracker(CustomerDataType.View))
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/view/trackViews.ts:215]: **💬 suggestion:** ‏you should not instanciate `createCustomerDataTrackerManager()` here. Either:
* don't use a `CustomerDataTracker` for the View context (for now). For that, you'll need to adjust `createContextManager` to make the argument optional.
* pass the manager [we got in `startRum`](https://github.com/DataDog/browser-sdk/blob/6682216f7fb02574b66dd03c1e8966c9504a1b3c/packages/rum-core/src/boot/startRum.ts#L62) down to `trackViews`.

I'm enclined to do the former. We can discuss about it if you want.

File: packages/rum-core/src/domain/view/trackViews.ts [hunk 3]
```diff
@@ -281,6 +313,7 @@ function newView(
     service,
     version,
     context,
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/view/trackViews.ts:315]: **💬 suggestion:** ‏Remove this `context` variable and property, and use `currentView.contextManager.getContext()` above (single source of truth, also right now `context` does not reflect the latest context of the view)

File: packages/rum-core/src/domain/view/trackViews.ts [hunk 4]
```diff
@@ -158,6 +169,18 @@ export function trackViews(
       currentView.end({ endClocks: startClocks })
       currentView = startNewView(ViewLoadingType.ROUTE_CHANGE, startClocks, options)
     },
+    setViewContext: (context: Context) => {
+      if (!isExperimentalFeatureEnabled(ExperimentalFeature.VIEW_SPECIFIC_CONTEXT)) {
+        return // TODO: remove this check when the feature is enabled by default
+      }
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/view/trackViews.ts:175]: **💬 suggestion:** ‏you could remove those experimental feature checks, as the "public api" methods are also behind this experimental feature, we can be sure the it won't be used here.

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**